crates/bitwarden-core/src/key_management/master_password.rs

```diff
@@ -0,0 +1,380 @@
+//! Mobile specific master password operations
+//!
+//! This module contains the data structures and error handling for master password unlock
+//! operations.
+
+use std::num::NonZeroU32;
+
+use bitwarden_api_api::models::{
+    master_password_unlock_response_model::MasterPasswordUnlockResponseModel, KdfType,
+};
+use bitwarden_crypto::{CryptoError, EncString, Kdf};
+use bitwarden_error::bitwarden_error;
+use serde::{Deserialize, Serialize};
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::*;
+
+use crate::{require, MissingFieldError};
+
+/// Error for master password related operations.
+#[allow(missing_docs)]
+#[bitwarden_error(flat)]
+#[derive(Debug, thiserror::Error)]
+pub enum MasterPasswordError {
+    #[error(transparent)]
+    Crypto(#[from] CryptoError),
+    #[error(transparent)]
+    MissingField(#[from] MissingFieldError),
+}
```

Comment:
```suggestion
#[bitwarden_error(flat)]
#[derive(Debug, thiserror::Error)]
pub enum MasterPasswordError {
    /// The wrapped encryption key could not be parsed
    #[error(transparent)]
    Crypto(#[from] CryptoError),
    /// The KDF data is malformed
    #[error(transparent)]
    MissingField(#[from] MissingFieldError),
}
```

I would actually suggest making these errors more descriptive. If I understand the rest of the file correctly, missingfield happens exactly when the KDF is broken, and Crypto happens exactly when the encstring parsing fails, so you could just skip wrapping the errors and make it more clear to the caller what actually failed:

```suggestion
#[bitwarden_error(flat)]
#[derive(Debug, thiserror::Error)]
pub enum MasterPasswordError {
    /// The wrapped encryption key could not be parsed because the encstring is malformed
    EncryptionKeyMalformed,
    /// The KDF data could not be parsed, because it is missing values or has an invalid value
    KdfMalformed
}
```

Reply:
Fixed with [9676069](https://github.com/bitwarden/sdk-internal/pull/376/commits/9676069424d62c6257e546f7fbf6d6f7445579dc)